pkg/odo/cli/storage/delete.go

```diff
@@ -29,97 +24,61 @@ var (
 `)
 )
 
-type StorageDeleteOptions struct {
+type DeleteOptions struct {
 	storageName            string
 	storageForceDeleteFlag bool
 	componentContext       string
 
-	isDevfile     bool
-	devfilePath   string
-	componentName string
 	*genericclioptions.Context
 }
 
-// NewStorageDeleteOptions creates a new StorageDeleteOptions instance
-func NewStorageDeleteOptions() *StorageDeleteOptions {
-	return &StorageDeleteOptions{devfilePath: component.DevfilePath}
+// NewStorageDeleteOptions creates a new DeleteOptions instance
+func NewStorageDeleteOptions() *DeleteOptions {
+	return &DeleteOptions{}
 }
 
-// Complete completes StorageDeleteOptions after they've been created
-func (o *StorageDeleteOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.devfilePath = filepath.Join(o.componentContext, o.devfilePath)
-	o.isDevfile = util.CheckPathExists(o.devfilePath)
-	if o.isDevfile {
-		o.Context = genericclioptions.NewDevfileContext(cmd)
+// Complete completes DeleteOptions after they've been created
+func (o *DeleteOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.Context, err = genericclioptions.New(genericclioptions.CreateParameters{
+		Cmd:              cmd,
+		DevfilePath:      component.DevfilePath,
+		ComponentContext: o.componentContext,
+	})
 
-		o.componentName = o.EnvSpecificInfo.GetName()
-	} else {
-		// This initializes the LocalConfigInfo as well
-		o.Context = genericclioptions.NewContext(cmd)
-
-		o.componentName = o.LocalConfigInfo.GetName()
+	if err != nil {
+		return err
 	}
+
 	o.storageName = args[0]
 	return
 }
 
-// Validate validates the StorageDeleteOptions based on completed values
-func (o *StorageDeleteOptions) Validate() (err error) {
-	if o.isDevfile {
-		return
-	}
-
-	exists := o.LocalConfigInfo.StorageExists(o.storageName)
-	if !exists {
+// Validate validates the DeleteOptions based on completed values
+func (o *DeleteOptions) Validate() (err error) {
+	gotStorage := o.LocalConfigProvider.GetStorage(o.storageName)
+	if gotStorage == nil {
 		return fmt.Errorf("the storage %v does not exists in the application %v, cause %v", o.storageName, o.Application, err)
 	}
 
 	return
 }
 
 // Run contains the logic for the odo storage delete command
-func (o *StorageDeleteOptions) Run() (err error) {
-	var deleteMsg string
-
-	var devFile devfileParser.DevfileObj
-	mPath := ""
-	if o.isDevfile {
-		devFile, err = devfile.ParseAndValidate(o.devfilePath)
-		if err != nil {
-			return err
-		}
-		err = validate.ValidateDevfileData(devFile.Data)
-		if err != nil {
-			return err
-		}
-		mPath, err = devFile.Data.GetVolumeMountPath(o.storageName)
-		if err != nil {
-			return err
-		}
-	} else {
-		mPath = o.LocalConfigInfo.GetMountPath(o.storageName)
+func (o *DeleteOptions) Run() (err error) {
+	mPath, err := o.Context.LocalConfigProvider.GetStorageMountPath(o.storageName)
```

Comment:
Thinking out loud - since this function's finding if the storage with the provided name exists, doesn't it make more sense to put it in `Validate`?

Reply:
> Thinking out loud - since this function's finding if the storage with the provided name exists, doesn't it make more sense to put it in Validate?

That function just gets the mount path for printing and doesn't perform any validation.